assets/scripts/datalayer/common.js

```diff
@@ -112,7 +112,10 @@ document.shopAnalytics = {
   $(onLoad);
 
   $(document)
-    .on('click', '.products .product a', onProductClick);
+    .on('click', '.products .product a', onProductClick)
+    .on('click', '.woocommerce-form-login button[name="login"]', onLoginFormSubmit)
+    .on('click', '.woocommerce-form-register button[name="register"]', onRegisterFormSubmit)
```

Comment:
If you hardcode the form element name attribute, this will not match the name used by my plugin.

Several options here:

* Is the event callback publicly exposed?  I.e., can I register `onLoginFormSubmit` from my external JS code?

* Is there a class name on these buttons that could be used instead?

* Do we need the `name` on the selector in the first place?  How many buttons are in each of these forms?

* If we need to be precise, can the `name` be pulled from a global context config setting?  Such as `window.shopAnalytics.event.click.login = 'login'`

Reply:
@sun What if I add some plugin configuration settings like:

![screenshot 2018-09-28 at 09 16 12](https://user-images.githubusercontent.com/13467494/46193657-38c9d300-c2ff-11e8-99f6-34979649bc07.png)

Would this work for you?

Reply:
No settings, please.  A filter is the maximum, because this needs to be code that refers to other code.  A database setting would be detached from the code.  That would essentially be option 4.

But what about the simpler options 2 and 3?  Not possible?

Reply:
> But what about the simpler options 2 and 3? Not possible?

I will look into that and answer ASAP.